src/pretix/control/views/item.py

```diff
@@ -283,6 +284,32 @@ def get_queryset(self):
         return self.request.event.questions.prefetch_related('items')
 
 
+@event_permission_required("can_change_items")
+def reorder_questions(request, organizer, event):
+    try:
+        ids = json.loads(request.body.decode('utf-8'))['ids']
+    except (JSONDecodeError, KeyError):
+        return HttpResponseBadRequest("expected JSON: {ids:[]}")
+
+    questions = request.event.questions.filter(id__in=ids)
+
+    if questions.count() != len(ids):
+        raise Http404(_("Some of the provided question ids are invalid."))
+
+    positions = questions.values_list('position', flat=True)
+
+    if positions.last() - positions.first() + 1 != questions.count():
+        return HttpResponseBadRequest("ids have to be from a consecutive range")
+
+    for pos, id in zip(positions, ids):
+        qt = questions.get(id=id)
```

Comment:
This creates a new database query on every iteration, could we pre-cache these in line 294?

Reply:
Ah, I'm not super familiar with Django ORM internals and assumed that the manager caches the result from line 294 and executes the query against that. Will fix.